core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/DetailedSampleBuilder.java

```diff
@@ -875,23 +882,28 @@ public DetailedSample build() {
 
     if (parent != null) {
       sample.setParent(parent);
-    } else if (!Identity.CATEGORY_NAME.equals(sampleClass.getSampleCategory())) {
-      Identity identity = new IdentityImpl();
-      if (externalName == null) {
-        throw new NullPointerException("Missing externalName");
+    } else {
+      DetailedSample parent = null;
+      int categoryIndex = CATEGORY_ORDER.indexOf(sampleClass.getSampleCategory());
+      if (categoryIndex < 0) {
+        throw new IllegalArgumentException("Sample has no parent and cannot infer order from sample category.");
       }
-      identity.setExternalName(externalName);
-      identity.setInternalName(internalName);
-      identity.setDonorSex(donorSex);
-      if (SampleTissue.CATEGORY_NAME.equals(sampleClass.getSampleCategory())) {
-        sample.setParent(identity);
-      } else {
-        if (tissueClass == null) throw new NullPointerException("Missing tissue class");
+      if (categoryIndex > 0 && externalName != null) {
```

Comment:
Instead of using indices explicitly, could this be refactored into a method like `needsParentCategory(String parentCategory)`?